modules/vaos/app/services/vaos/middleware/response/errors.rb

```diff
@@ -9,49 +9,53 @@ class Errors < Faraday::Response::Middleware
         def on_complete(env)
           return if env.success?
 
-          Raven.extra_context(vamf_status: env.status, vamf_body: env.body, vamf_url: env.url)
+          @status = env.status
```

Comment:
since these aren't used as accessors, could these values be passed into the methods that use them instead of being part of the state?

Reply:
it was a lot of passing around, yes, we could switch to that but I thought this looked a little cleaner. We could also make them accessors but does that violate some rule in ruby? I don't know if it does or not but `on_complete` is not really the same thing as `initialize` so maybe to your point passing them as reference is the way to go.

Reply:
I'm a beginner at Ruby, but in our case, I don't think that's going to work.  We have an additional method that uses those accessors (I call them class member variables).
```
def response_values
   {
     detail: detail,
     source: { vamf_url: @url, vamf_body: @body, vamf_status: @status }
   }
end
```

If we were to inject `env` directly into that method, we would have to update the calls `response_values` to pass that as well.  Which we could do, if that's what you're recommending.

Reply:
nm, it's a private method